src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs

```diff
@@ -754,6 +754,34 @@ private void TimerCallback(object backupTaskDetails)
             CreateBackupTask(periodicBackup, backupDetails);
         }
 
+        public async Task RunAdhocBackup(int taskId, bool isFullBackup)
+        {
+            var dr = GetDatabaseRecord();
+
+            PeriodicBackupConfiguration config = null;
+            // ReSharper disable once ForCanBeConvertedToForeach
+            for (int i = 0; i < dr.PeriodicBackups.Count; i++)
+            {
+                if (dr.PeriodicBackups[i].TaskId != taskId)
+                    continue;
+
+                config = dr.PeriodicBackups[i];
+                break;
+            }
+
+            if (config == null) //didn't find relevant backup task...
+                return;
+
+            if (_periodicBackups.TryGetValue(taskId, out var periodicBackup) == false ||
+                periodicBackup.Disposed)
+            {
+                // backup task doesn't exist anymore or is canceled
```

Comment:
Not good, this is an error, and should be reported as such, otherwise user may believe that the backup happened.